### PR TITLE
Update property model

### PR DESCRIPTION
**Pull request by galvana**
Closes [PROD-2025](https://ethyca.atlassian.net/browse/PROD-2025)

### Description Of Changes

Adds `privacy_center_config`, `stylesheet`, and `paths` to the `Property` model.

### Code Changes

* [ ] Expanded the `PrivacyCenterConfig` schema to fully implement the schema as it is defined in the `privacy_center`  client
* [ ] Moved the CSS validator from fidesplus into fides and added a new `CssStr` custom type with CSS validation
* [ ] Added `privacy_center_config` and `stylesheet` fields to the `plus_property` table
* [ ] Added `plus_property_path` table to track the Property paths

### Steps to Confirm

* [ ] See testing steps in https://github.com/ethyca/fidesplus/pull/1423

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [ ] Update `CHANGELOG.md`
* [ ] For API changes, the [Postman collection](https://github.com/ethyca/fides/blob/main/docs/fides/docs/development/postman/Fides.postman_collection.json) has been updated

[PROD-2025]: https://ethyca.atlassian.net/browse/PROD-2025?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/fides/api/schemas/privacy_center_config.py [hunk 1]
```diff
@@ -28,16 +51,85 @@ def validate_default_value(cls, values: Dict[str, Any]) -> Dict[str, Any]:
...
+
+class ConsentConfigPage(FidesSchema):
+    consentOptions: List[ConfigConsentOption]
+    description: str
+    description_subtext: Optional[List[str]]
+    policy_key: Optional[str]
+    title: str
+
+    @root_validator(pre=True)
+    def validate_consent_options(cls, values: Dict[str, Any]) -> Dict[str, Any]:
```
Comment (adamsachs) [inline src/fides/api/schemas/privacy_center_config.py:100]: again, i think this could be a bit more defensive given that it's a `pre` validator. i realize that the values here may be constrained by the FE client, but better to make as few assumptions as possible about the input values to keep the API robust.

File: tests/ops/models/test_property.py [hunk 2]
```diff
@@ -29,24 +42,62 @@ def minimal_experience(
...
+                    name="Second Property",
+                    type=PropertyType.website,
+                    experiences=[minimal_experience],
+                    privacy_center_config=privacy_center_config,
+                    stylesheet=":root:root { --chakra-colors-gray-50: #fff9ea; }",
+                    paths=["test"],
+                ).dict(),
+            )
+
+        first_prop.delete(db)
```
Comment (adamsachs) [inline tests/ops/models/test_property.py:99]: haven't verified this test specifically, but from what i can see with testing on the API, i think the `Second Property` actually gets created here _before_ the `IntegrityError` is raised, so i think this test could/should be adjusted to ensure that the `Second Property` isn't actually created!

**Review by adamsachs — CHANGES_REQUESTED** [review]
@galvana this generally looks on the right track, but i've got some points that i think would be good to address to make these models and the API a bit more robust. let me know your thoughts or any questions!

File: src/fides/api/models/property.py [hunk 3]
```diff
@@ -60,21 +73,62 @@ def create(
...
         experiences = data.pop("experiences", [])
+        paths = data.pop("paths", None) or []
+
         prop: Property = super().create(db=db, data=data, check_name=check_name)
         link_experience_configs_to_property(
             db, experience_configs=experiences, prop=prop
         )
-        return prop
+
+        property_paths = [
```
[the remark this answers is not in the document]
Comment (galvana) [reply src/fides/api/models/property.py:83]: Checking existing `PropertyPaths` and raises an exception if the paths are already associated with another Property. Updated tests.

File: src/fides/api/models/property.py [hunk 4]
```diff
@@ -60,21 +73,62 @@ def create(
...
+
+        # delete PropertyPath instances that are not in the updated paths
+        for path in set(existing_paths.keys()) - updated_paths:
+            db.delete(existing_paths[path])
+
+        # create new PropertyPath instances for paths that don't exist
+        for path in updated_paths - set(existing_paths.keys()):
+            self._property_paths.append(PropertyPath(property_id=self.id, path=path))
+
+        self.save(db)
```
[the remark this answers is not in the document]
Comment (galvana) [reply src/fides/api/models/property.py:110]: Similar fix as above

File: tests/ops/models/test_property.py (hunk 2 above)
Comment (galvana) [reply tests/ops/models/test_property.py:99]: Updated test

File: src/fides/api/models/property.py [hunk 5]
```diff
@@ -60,21 +74,82 @@ def create(
...
+        paths = data.pop("paths", None) or []
+
+        matching_paths = PropertyPath.filter(
+            db=db, conditions=(PropertyPath.path.in_(paths))
+        ).all()
+        if matching_paths:
+            raise KeyOrNameAlreadyExists(
+                f'The path(s) \'{", ".join([matching_path.path for matching_path in matching_paths])}\' are already associated with another property.'
+            )
+
```
Comment (adamsachs) [inline src/fides/api/models/property.py:86]: very nice! as a nit, i might move this into a helper function to share between create and update in case the logic ever needs to be adjusted, but i'll let you decide whether or not to add that last bit of polish :)

File: src/fides/api/schemas/property.py [hunk 6]
```diff
@@ -29,3 +33,10 @@ class Property(FidesSchema):
     type: PropertyType
     id: Optional[str] = None
     experiences: List[MinimalPrivacyExperience]
+    privacy_center_config: Optional[PrivacyCenterConfig]
+    stylesheet: Optional[CssStr]
+    paths: List[str]
```
[the remark this answers is not in the document]
Comment (adamsachs) [reply src/fides/api/schemas/property.py:38]: ah ok! makes sense, thanks for clarifying.

File: tests/ops/models/test_property.py [hunk 7]
```diff
@@ -29,43 +45,92 @@ def minimal_experience(
...
+                    privacy_center_config=privacy_center_config,
+                    stylesheet=":root:root { --chakra-colors-gray-50: #fff9ea; }",
+                    paths=["test"],
+                ).dict(),
+            )
+
+        second_prop = Property.filter(
+            db=db, conditions=(Property.name == "Second Property")
+        ).first()
+        assert second_prop is None
```
Comment (adamsachs) [inline tests/ops/models/test_property.py:105]: 🎊

File: tests/ops/schemas/test_property_schema.py
Comment (adamsachs) [inline tests/ops/schemas/test_property_schema.py:1]: these are great! nice job covering these edge cases 👍

**Review by adamsachs — APPROVED** [review]
this is looking good to me @galvana! thanks for addressing the feedback 👍
